### ISSUE TITLE
Feature Request: "disableUserEditable" but only after creation

### ISSUE THREAD

**Issue by rvlasveld** [lolopinto/ent#1331]
Currently a field can have the option `disableUserEditable`. When enabled, and a `defaultValueOnCreate` is provided, no setters in the action or graphql mutation will be created. 

A common scenario is that a field can only be set on creation, but never be changed after that. This can currently be enforced through validators or by excluding the fields in the action definition. It could be convenient to have this as a configuration on the field. 

Suggestion: "immutable"? That could work for either user-provided or "default on create" values. 

The "...UserEditable" also suggests that it relates to _editing_, while it also has effect _creating_. Perhaps it should be renamed to something like "userProvided"?

**Comment by lolopinto** [comment]
this seems like the same as this https://github.com/lolopinto/ent/issues/572, right?

`immutable` makes sense

**Comment by lolopinto** [comment]
do we also prevent it from being settable in builders? generally, i like everything to be settable in builders so that developers still have the ability to do anything they want e.g. an internal tool changing this

**Comment by rvlasveld** [comment]
Ah yeah, seems the same as #572.

I think there are certain fields, such as a creatorID, which should be
immutable on the lowest level of the ent framework. Any overrides should in
that case happen directly in SQL. Ideally the ent framework should prevent
a developer from making changes that go against the data model rules (such
as changing a creator). It could be different from permission-based fields;
e.g. changing an "owner" if the viewer has sufficient permissions to do so.

**Comment by lolopinto** [comment]
hmm, not sure if i like that. will have to think about that.

I think that's why I ended up with `userEditable` vs other names because it's not editable by the end facing user but the developer can still do whatever they want 

or in actions with a schema like the following:

```ts
{
  excludedFields: ['bar'],
  operation: ActionOperation.Edit,
}

```

you can still do something like:

```ts
const action = EditFooAction.create(vc, foo, input);
action.builder.updateInput({bar: 'newValue'});
await action.saveX();
```

or update in a trigger as needed

**Comment by rvlasveld** [comment]
Yeah I think those are okay alternatives. I'm mainly thinking from a
situation where there are many developers, and not everyone might know the
data model rules / assumptions. I would like to prevent that someone, in a
trigger, sets a value which someone else decided to be immutable. Although
perhaps the two last alternatives are enough signal that someone is working
around the main assumptions.

**Comment by lolopinto** [comment]
maybe `immutable` is strong enough that the builder API is `overrideCreatorId(newCreator)` and it doesn't show up in builder so it's clearer what's happening and someone is making a conscious decision to override.

```ts
const action = EditFooAction.create(vc, foo, input);
action.builder.updateInput({creatorId: newCreator}); // doesn't work
action.builder.overrideCreatorId(newCreator); // works
```
